src/jloptions.c

```diff
@@ -815,6 +819,14 @@ JL_DLLEXPORT void jl_parse_opts(int *argcp, char ***argvp)
             if (jl_options.heap_size_hint == 0)
                 jl_errorf("julia: invalid argument to --heap-size-hint without memory size specified");
 
+            break;
+        case opt_gc_threads:
+            if (optarg != NULL) {
+                jl_options.ngcthreads = atoi(optarg);
+            }
+            if (jl_options.ngcthreads == 0)
+                jl_errorf("julia: invalid argument to --gcthreads without number of threads specified");
+
```

Comment:
```suggestion
            errno = 0;
            long ngcthreads = strtol(optarg, &endptr, 10);
            if (errno != 0 || optarg == endptr || *endptr != 0 || ngcthreads < 1 || ngcthreads >= INT8_MAX)
                jl_errorf("julia: --gcthreads=<n>; n must be an integer >= 1");
            jl_options.ngcthreads = (int8_t)ngcthreads;
```

like: https://github.com/JuliaLang/julia/blob/c6ed7d7c54a501090019e2f509935f6f21bdf9af/src/jloptions.c#L512-L523